### PR TITLE
[SPARK-35083][CORE][FOLLLOWUP] Improve docs and migration guide

### PR DESCRIPTION
**Pull request by ulysses-you**
### What changes were proposed in this pull request?

* improve docs in `docs/job-scheduling.md`
* add migration guide docs in `docs/core-migration-guide.md`

### Why are the changes needed?

Help user to migrate.

### Does this PR introduce _any_ user-facing change?

yes

### How was this patch tested?

Pass CI

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]
Looks good

**Comment by HyukjinKwon** [comment]
cc @gengliangwang and @Ngone51 FYI

**Review by gengliangwang — APPROVED** [review]

**Review by Ngone51 — APPROVED** [review]

**Comment by ulysses-you** [comment]
thank you all for the approved, also FYI @dongjoon-hyun

File: docs/job-scheduling.md [hunk 1]
```diff
@@ -254,10 +254,13 @@ properties:
 
 The pool properties can be set by creating an XML file, similar to `conf/fairscheduler.xml.template`,
 and either putting a file named `fairscheduler.xml` on the classpath, or setting `spark.scheduler.allocation.file` property in your
-[SparkConf](configuration.html#spark-properties). The file path can either be a local file path or HDFS file path.
+[SparkConf](configuration.html#spark-properties). The file path respect the hadoop configuration and can either be a local file path or HDFS file path.
```
Comment (MaxGekk) [inline docs/job-scheduling.md:257]: path respect -> path respects

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master and 3.2

**Pull request closed without merging**